internal/pkg/cli/env_delete.go

```diff
@@ -346,6 +369,78 @@ func (o *deleteEnvOpts) ensureRolesAreRetained() error {
 	return nil
 }
 
+// emptyBuckets returns nil if buckets were deleted successfully. Otherwise, returns the error.
+func (o *deleteEnvOpts) emptyBuckets() error {
+	s3buckets, err := o.rg.GetResources(&resourcegroupstaggingapi.GetResourcesInput{
+		ResourceTypeFilters: aws.StringSlice([]string{"s3:bucket"}),
+		TagFilters: []*resourcegroupstaggingapi.TagFilter{
+			{
+				Key:    aws.String(stack.EnvS3BucketStackNameTagKey),
+				Values: []*string{aws.String(stack.NameForEnv(o.appName, o.name))},
+			},
+			{
+				Key:    aws.String(stack.EnvS3BucketLogicalIDTagKey),
+				Values: aws.StringSlice(stack.EnvManagedS3BucketLogicalIds),
+			},
+			{
+				Key:    aws.String(deploy.EnvTagKey),
+				Values: []*string{aws.String(o.name)},
+			},
+			{
+				Key:    aws.String(deploy.AppTagKey),
+				Values: []*string{aws.String(o.appName)},
+			},
+		},
+	})
+
+	if err != nil {
+		return fmt.Errorf("find s3 bucket resources: %w", err)
+	}
+
+	envResources, err := o.envStackDescriber.Resources()
+	if err != nil {
+		return fmt.Errorf("find stack resources: %w", err)
+	}
+
+	var stackBucketARNs []string
+	for _, resource := range envResources {
+		if contains(resource.PhysicalID, stack.EnvManagedS3BucketLogicalIds) {
```

Comment:
as far as I remember the `PhysicalID`s are _typically_ the ARN of the resource, not the logical IDs. Maybe you meant to do this ⬇️ ?

```suggestion
		if contains(resource.LogicalID, stack.EnvManagedS3BucketLogicalIds) {
```

Reply:
Thanks for the catch on this! I think I accidentally changed it to `PhysicalID` when refactoring this section at some point!